src/contracts/libraries/GPv2Encoding.sol

```diff
@@ -324,6 +325,94 @@ library GPv2Encoding {
         }
     }
 
+    /// @dev Decodes an interaction from calldata into memory.
+    ///
+    /// An encoded interaction has three components: the target address, the data
+    /// length, and the actual interaction data of variable size.
+    ///
+    /// ```
+    /// struct EncodedInteraction {
+    ///     address target;
+    ///     uint24 dataLength;
+    ///     bytes callData;
+    /// }
+    /// ```
+    ///
+    /// All entries are tightly packed together in this order in the encoded
+    /// calldata. Example:
+    ///
+    /// input:    0x73c14081446bd1e4eb165250e826e80c5a523783000010000102030405060708090a0b0c0d0e0f
+    /// decoding:   [...............target.................][leng][............data..............]
+    /// stride:                                          20     3    (defined in length field) 16
+    ///
+    /// This function enforces that the encoded data stores enough bytes to
+    /// cover the full length of the decoded interaction.
+    ///
+    /// The size of `dataLength` limits the maximum calldata that can be used in
+    /// an interaction. Based on the current rules of the Ethereum protocol,
+    /// this length is enough to include any valid transaction: an extra
+    /// calldata byte costs at least 4 gas, and the maximum gas spent in a block
+    /// is 12.5M. This gives an upper bound on the calldata that can be included
+    /// in a block of
+    ///   3.125.000 < 16.777.216 = 2**(3*8) .
+    ///
+    /// @param encodedInteractions The interactions as encoded calldata bytes.
+    /// @param interaction The memory location to decode the interaction to.
+    /// @return reducedEncodedInteractions The part of encodedInteractions that
+    /// has not been decoded after this function is executed.
+    function decodeInteraction(
+        bytes calldata encodedInteractions,
+        Interaction memory interaction
+    ) internal pure returns (bytes calldata reducedEncodedInteractions) {
```

Comment:
naming nit: `remainingEncodedInteractions`? I find reducing to be a bit of an overloaded term.

Reply:
I am not sure I understand how we are decoding an interaction if we are just returning the n-1 interactions that remain undecoded?

Reply:
The decoded interaction gets written to `Interaction memory interaction`.